tests/unit/test_errorfactory.py

```diff
@@ -0,0 +1,30 @@
+import sys
+
+from nose.tools import assert_equals, assert_true
+
+from botocore.exceptions import ClientError
+from botocore.errorfactory import ServiceErrorFactory
+from botocore.model import ServiceModel
+
+
+def test_errorfactory():
+    model = {
+        'operations': {
+            'OperationName': {
+                'name': 'OperationName',
+                'errors': [{'shape': 'NoSuchResourceException'}],
+            }
+        },
+        'shapes': {
+            'NoSuchResourceException': {
+                'type': 'structure',
+                'members': {}
+            }
+        }
+    }
+    factory = ServiceErrorFactory(ServiceModel(model))
+    members = dir(factory)
+    if sys.version_info >= (2, 7):
```

Comment:
I don't have a Python 2.6 runtime handy and the tests take a while to run so I'm at a loss. Here is the error from Travis:
```
======================================================================
FAIL: tests.unit.test_errorfactory.test_errorfactory
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/virtualenv/python2.6.9/lib/python2.6/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/travis/build/boto/botocore/tests/unit/test_errorfactory.py", line 24, in test_errorfactory
    assert_equals(dir(factory), ['NoSuchResourceException'])
AssertionError: ['__dir__', '__doc__', '__getattr__', '__init__', '__module__', '_error_shapes', '_service_model'] != ['NoSuchResourceException']

----------------------------------------------------------------------
```